Userland/Libraries/LibGfx/Matrix4x4.h

```diff
@@ -37,16 +37,6 @@ constexpr static Vector3<T> transform_point(const Matrix4x4<T>& m, const Vector3
         p.x() * elements[2][0] + p.y() * elements[2][1] + p.z() * elements[2][2] + elements[2][3]);
 }
 
-template<typename T>
```

Comment:
Why was this removed?

Reply:
Its only usage was removed, and I'd imagine this function was very OpenGL-specific. If you think there is a good remaining use case I'll leave it in.

Reply:
Also, I don't think the original code was correct? It transforms the normals with the model view transform, but we should apply the inverse transformation instead. That is what `normal_transform` is for.

Reply:
It the counterpart to `transform_point()`.
transform_point uses an implied w coordinate of 1 and transform_direction uses an implied w coordinate of 0. Very handy when you want to multiply vec3s by a 4x4 matrix which would otherwise require constructing a vec4 first.

Reply:
As discussed on Discord; we believe that `transform_direction` is a too specialized case to keep. I've tried to generalize things a bit by adding `Matrix::submatrix_from_top_left<N>()`. Let me know what you think!

Reply:
Yeah looks good. Maybe add a fixme to transform_point if you son't want to remove this right now.